cmd/azure.go

```diff
@@ -38,9 +39,35 @@ import (
 
 const globalAzureAPIVersion = "2016-05-31"
 
+// To store metadata during NewMultipartUpload which will be used after
+// CompleteMultipartUpload to call SetBlobMetadata.
+type azureMultipart struct {
+	meta map[string]map[string]string
+	mu   sync.Mutex
```

Comment:
since this lock is for protecting the map it would not be right to force fit nslock here.
have a look at the get API:
```
// To store metadata during NewMultipartUpload which will be used after                                                                     
// CompleteMultipartUpload to call SetBlobMetadata.                                                                                         
type azureMultipartMetaInfo struct {                                                                                                        
        meta map[string]map[string]string                                                                                                   
        sync.Mutex                                                                                                                          
}                                                                                                                                           
                                                                                                                                            
func (a *azureMultipartMetaInfo) get(key string) map[string]string {                                                                        
        a.Lock()                                                                                                                            
        defer a.Unlock()                                                                                                                    
        return a.meta[key]                                                                                                                  
}
```
this will look like:
```
// To store metadata during NewMultipartUpload which will be used after                                                                     
// CompleteMultipartUpload to call SetBlobMetadata.                                                                                         
type azureMultipartMetaInfo struct {                                                                                                        
        meta map[string]map[string]string                                                                                                   
}                                                                                                                                           
                                                                                                                                            
func (a *azureMultipartMetaInfo) get(key, bucket, object string) map[string]string {                                                                        
        objectLock := globalNSMutex.NewNSLock(bucket, object)                                                                               
        objectLock.RLock()                                                                                                                  
        defer objectLock.RUnlock()
        return a.meta[key]                                                                                                                  
}
```